src/components/content/credential/AddCredential.tsx

```diff
@@ -40,14 +41,84 @@ function AddCredential({
     const [tipMessage, setTipMessage] = useState<string>('');
     const [tipType, setTipType] = useState<'error' | 'success' | undefined>(undefined);
     const oidcToken: OidcIdToken = useOidcIdToken();
+
+    const credentialTypesQuery = useQuery({
+        queryKey: ['credentialTypesQuery', currentCsp],
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        queryFn: () => CredentialsManagementService.getCredentialTypesByCsp(currentCsp!),
+        staleTime: 60000,
+        enabled: currentCsp !== undefined,
+    });
+
+    useEffect(() => {
+        const types = credentialTypesQuery.data;
+        if (types !== undefined && types.length > 0) {
+            setCredentialTypeList(types as CredentialVariables.type[]);
+        } else {
+            setCredentialTypeList([]);
+        }
+    }, [credentialTypesQuery.data, credentialTypesQuery.isSuccess]);
+
+    useEffect(() => {
+        setCredentialTypeList([]);
+    }, [credentialTypesQuery.error]);
+
+    const credentialCapabilitiesQuery = useQuery({
```

Comment:
same also for the dropdowns dependent on this query. Please show 'loading' when data is being loaded.